### DISCUSSION TITLE
Why "Access your data for all websites" as optional permission?

### DISCUSSION THREAD

**Discussion by Gibdos** [floccusaddon/floccus#1514] [Q&A]
Today I saw that Floccus wanted a new permission. When I went to check it wanted the aforementioned "Access your data for all websites" as an optional permission. Just curious why it would ever need that permission to work as a basic bookmark sync extension.

According to Mozilla that permission allows Floccus
`The extension could read the content of any web page you visit, as well as data you enter into those web pages, such as usernames and passwords. `

Firefox is 121 and Floccus is 5.0.7

**Comment by marcelklehr** [comment]
The reason you get asked is because of a manifest version change, but in reality, floccus has always had this permission already. It is necessary for accessing your self-hosted server. This cannot be limited, because everybody's server has a different URL. Unfortunately, the way webextensions work currently, floccus also gets access to all the data the browser has collected on those websites. However, floccus makes no use of that data and doesn't in any way collect information about you. Also see the [guide: Install in your browser](https://floccus.org/guides)
